metrics/sources/manager.go

```diff
@@ -96,21 +96,23 @@ func (this *sourceManager) ScrapeMetrics(start, end time.Time) *DataBatch {
 			time.Sleep(time.Duration(rand.Intn(delayMs)) * time.Millisecond)
 
 			glog.V(2).Infof("Querying source: %s", source)
-			metrics := scrape(source, start, end)
-			now := time.Now()
-			if !now.Before(timeoutTime) {
-				glog.Warningf("Failed to get %s response in time", source)
-				return
-			}
-			timeForResponse := timeoutTime.Sub(now)
-
-			select {
-			case channel <- metrics:
-				// passed the response correctly.
-				return
-			case <-time.After(timeForResponse):
-				glog.Warningf("Failed to send the response back %s", source)
-				return
+			metrics, err := scrape(source, start, end)
+			if err == nil {
```

Comment:
again, `if err != nil`:

```go
if err != nil {
    glog.Errorf("error scraping source: %v")
    continue
```

make sure to log the error, and not just discard it